### ISSUE TITLE
I installed the OpenCV package through vcpkg, but CMAKE is indicating that it can't be found

### ISSUE THREAD

**Issue by zanpocc** [microsoft/vcpkg#30647]
My operating system is Mac m1 13.3, and I am using the latest code for vcpkg. 
I installed the OpenCV package through vcpkg, 

```
./vcpkg install opencv
Computing installation plan...
The following packages are already installed:
    opencv[core,default-features]:arm64-osx -> 4.7.0#1
opencv:arm64-osx is already installed
Restored 0 package(s) from /Users/xxxx/.cache/vcpkg/archives in 1.54 us. Use --debug to see more details.
Total install time: 91 us
opencv provides CMake targets:

    # this is heuristically generated, and may not be correct
    find_package(OpenCV CONFIG REQUIRED)
    # note: 12 additional targets are not displayed.
    target_link_libraries(main PRIVATE opencv_ml tegra_hal opencv_dnn opencv_core)
```

and then followed the prompts in CMake to import it. 

```
set(VCPKG_TARGET_TRIPLET ${OS_ARCH})
set(CMAKE_TOOLCHAIN_FILE ${VCPKG_PATH}/scripts/buildsystems/vcpkg.cmake)

set(CMAKE_CXX_STANDARD 17)

find_package(OpenCV CONFIG REQUIRED)
target_link_libraries(Rasterizer PRIVATE opencv_ml tegra_hal opencv_dnn opencv_core)
```

However, CMake is indicating that it can't be found. 

```
CMake Error at CMakeLists.txt:39 (find_package):
  Could not find a package configuration file provided by "OpenCV" with any
  of the following names:

    OpenCVConfig.cmake
    opencv-config.cmake

  Add the installation prefix of "OpenCV" to CMAKE_PREFIX_PATH or set
  "OpenCV_DIR" to a directory containing one of the above files.  If "OpenCV"
  provides a separate development package or SDK, be sure it has been
  installed.

```

But other packages installed through vcpkg seem to be working fine.
What should I do?

**Comment by MonicaLiu0311** [comment]
```
vcpkg@VCPKGMM-12 build % /Users/vcpkg/Monica/vcpkg/downloads/tools/cmake-3.25.1-osx/cmake-3.25.1-macos-universal/CMake.app/Contents/bin/cmake ..
-- The C compiler identification is AppleClang 12.0.5.12050022
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Applications/Xcode.app/Contents/Developer/Toolchains/XcodeDefault.xctoolchain/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Found Protobuf: /Users/vcpkg/Monica/vcpkg/installed/x64-osx/tools/protobuf/protoc (found version "3.21.12.0")
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD
-- Performing Test CMAKE_HAVE_LIBC_PTHREAD - Success
-- Found Threads: TRUE
-- Found TIFF: /Users/vcpkg/Monica/vcpkg/installed/x64-osx/lib/libtiff.a (found version "4.5.0")
-- Found LibLZMA: /Users/vcpkg/Monica/vcpkg/installed/x64-osx/lib/liblzma.a (found version "5.4.1")
-- Found JPEG: /Users/vcpkg/Monica/vcpkg/installed/x64-osx/lib/libjpeg.a (found version "62")
-- Found ZLIB: /Users/vcpkg/Monica/vcpkg/installed/x64-osx/lib/libz.a (found version "1.2.13")
-- Found OpenCV: /Users/vcpkg/Monica/vcpkg/installed/x64-osx (found version "4.7.0")
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/vcpkg/Monica/CMakeFindUsage/build
```
Is there something wrong with your CMAKE_TOOLCHAIN_FILE path? I can find it.
```
  1 cmake_minimum_required(VERSION 3.8)
  2
  3 set(CMAKE_TOOLCHAIN_FILE "/Users/vcpkg/Monica/vcpkg/scripts/buildsystems/vcpkg.cmake")
  4
  5 project("CMakeFindUsage")
  6
  7 add_executable(main "CMakeFindUsage.cpp")
  8
  9 find_package(OpenCV CONFIG REQUIRED)
 10 target_link_libraries(main PRIVATE opencv_ml tegra_hal opencv_dnn opencv_core)
```

**Comment by zanpocc** [comment]
@MonicaLiu0311
I did some testing and found that the issue wasn't with my CMAKE_TOOLCHAIN_FILE path. 
It was actually because I placed `project(my_project)` above `set(CMAKE_TOOLCHAIN_FILE ${VCPKG_PATH}/scripts/buildsystems/vcpkg.cmake)`. 
This caused my packages to go missing. But, when I moved `project(my_project)` below that line and deleted CMakeCache.txt, things worked fine. 

It was confusing because the official docs didn't mention this. And using cmake -DCMAKE_TOOLCHAIN_FILE worked without any problems.
